boa_engine/src/builtins/object/mod.rs

```diff
@@ -201,6 +202,52 @@ impl Object {
         Ok(JsValue::undefined())
     }
 
+    /// `Object.prototype.__lookupGetter__(prop)`
+    ///
+    /// Returns the function bound as a getter to the specified property.
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///  - [MDN documentation][mdn]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-object.prototype.__lookupGetter__
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/__lookupGetter__
+    pub fn legacy_lookup_getter(
+        this: &JsValue,
+        args: &[JsValue],
+        context: &mut Context,
+    ) -> JsResult<JsValue> {
+        // 1. Let O be ? ToObject(this value).
+        let mut obj = this.to_object(context)?;
+
+        // 2. Let key be ? ToPropertyKey(P).
+        let key = args.get_or_undefined(0).to_property_key(context)?;
+
+        // 3. Repeat
+        loop {
+            // a. Let desc be ? O.[[GetOwnProperty]](key).
+            let desc = obj.__get_own_property__(&key, context)?;
+
+            // b. If desc is not undefined, then
+            if let Some(current_desc) = desc {
+                // i. If IsAccessorDescriptor(desc) is true, return desc.[[Get]].
+                return if current_desc.is_accessor_descriptor() {
+                    Ok(current_desc.get().expect("get cannot fail here").into())
```

Comment:
Instead of `.get().expect()` you can use the `.expect_get()` here. That method is build exactly for such a case.

Reply:
Hadn't spotted that convenvience method - updated.